### ISSUE TITLE
Can't have concatenated string macros

### ISSUE THREAD

**Issue by elbrujohalcon** [AdRoll/rebar3_format#22]
This code breaks the formatter:

```erlang
-module(x).
-export([y/0]).
-define(STRING, "a comment that's a string").
y() -> ?STRING ?STRING.
```

The error report looks like `Formatting error: syntax error before: '? STRING'.`

This is a well-known situation in OTP. They actually have a test for it [here](https://github.com/erlang/otp/blob/dda57a83e920167bad719aea05bbb4746b566903/lib/syntax_tools/test/syntax_tools_SUITE_data/syntax_tools_test.erl#L51).

**Comment by elbrujohalcon** [comment]
In fact, that same file also showcases this nasty case…
```erlang
-define(attrib, some_attrib).

-?attrib([foo2/2]).
```
Which also produces an error:
```erlang
{function_clause,
    [{erl_recomment,leaf_node_value,
         [{tree,10,10,macro,
              {attr,10,[],none},
              [],[],
              [{list,10,10,
                   [{leaf,10,10,[],[],
                        {tree,atom,{attr,10,[],none},attrib}}]}]}],
         [{file,"erl_recomment.erl"},{line,711}]},
     {erl_recomment,check_file_attr_1,2,
         [{file,"erl_recomment.erl"},{line,282}]},
     {erl_recomment,filter_forms,3,[{file,"erl_recomment.erl"},{line,229}]},
     {erl_recomment,filter_forms,3,[{file,"erl_recomment.erl"},{line,253}]},
     {erl_recomment,recomment_forms,3,[{file,"erl_recomment.erl"},{line,136}]},
     {rebar3_formatter,format,4,
         [{file,
              "/Users/fernandobenavides/Projects/Semantic-Sugar/rebar3_format/src/rebar3_formatter.erl"},
          {line,54}]},
     {lists,foreach,2,[{file,"lists.erl"},{line,1338}]},
     {rebar3_format_prv,format,2,
         [{file,
              "/Users/fernandobenavides/Projects/Semantic-Sugar/rebar3_format/src/rebar3_format_prv.erl"},
          {line,76}]}]}
```

**Comment by elbrujohalcon** [comment]
And another macro-related thingy… hopefully solvable with the same solution as the other 2…
```erlang
-define(OUT_OF_CONTEXT_CLAUSE, (true) -> false).

in_context?OUT_OF_CONTEXT_CLAUSE.
```

**Comment by elbrujohalcon** [comment]
Let's try `ktn_dodger:parse_file(File, [{clever, true}])` for this.

**Comment by elbrujohalcon** [comment]
This one is probably unsolvable. Things like `in_context?OUT_OF_CONTEXT_CLAUSE` are basically impossible to parse if you don't pre-process macros.

**Comment by elbrujohalcon** [comment]
2 out of 3 issues solved already with #207. The remaining one went to #211.

**Comment by elbrujohalcon** [comment]
> This one is probably unsolvable. Things like `in_context?OUT_OF_CONTEXT_CLAUSE` are basically impossible to parse if you don't pre-process macros.

As the wise folks of the OTP team [once said](https://github.com/erlang/otp/blob/a819a334e8162a7501ca7e2bde3f22c9fa612125/lib/edoc/src/edoc.erl#L763-L767)…
> If the error is caused by too exotic macro definitions or uses of macros, adding option `{preprocess, true}` can help…

But we don't have that option here, so… 🤷‍♂️
